components/script/dom/abortsignal.rs

```diff
@@ -48,6 +51,15 @@ pub(crate) struct AbortSignal {
 
     /// <https://dom.spec.whatwg.org/#abortsignal-abort-algorithms>
     abort_algorithms: RefCell<Vec<AbortAlgorithm>>,
+
+    /// <https://dom.spec.whatwg.org/#abortsignal-dependent>
+    dependent: Cell<bool>,
+
+    /// <https://dom.spec.whatwg.org/#abortsignal-source-signals>
+    source_signals: DomRefCell<IndexSet<DomRoot<AbortSignal>>>,
```

Comment:
Ran into several issues related to rooting with these. I ended up making this `DomRoot` so that the `rooted_vec` on line 117 works. I am not sure if this is what we should be doing though, as I still don't fully grasp rooting.

Reply:
We should be able to store `Dom<AbortSignal>` in these fields.

Reply:
Thanks!

Reply:
Hm back with the original error:

```
error: Expression of type indexmap::set::Iter<'{erased}, script_bindings::root::Dom<dom::abortsignal::AbortSignal>> must be rooted.
```

I ran `--use-crown` locally, not sure why it doesn't catch that. How can I fix this without storing DomRoot in the field?

Reply:
> Is it possible to iterate over `&AbortSignal` instead of `Dom<AbortSignal>` instead?

I think it should be by using .map

Reply:
Need to move the `as_rooted()` into the `map` and then crown was happy.

Reply:
Hm, maybe we should just allow all types that implements `Iterator`? We already have mechanism to check if type implements trait: https://github.com/servo/servo/blob/64115c6197e04de205907981265b41f5dd128936/support/crown/src/trace_in_no_trace.rs#L110

Reply:
I think it would be enough for this PR to simply do `#[allow(crown::unrooted_must_root)]` and fix crown properly in a follow up.

Reply:
Filed #39333 for this and updating the code with the ignore.